R/datashield.assign.R

```diff
@@ -364,13 +370,19 @@ datashield.assign.resource <- function(conns, symbol, resource, async=TRUE, succ
         success(conns@name)
       }
     }, error = function(e) {
-      .appendError(conns@name, e$message)
+      .appendError(conns@name, conditionMessage(e))
       if (.is.callback(error)) {
-        error(conns@name, e$message)
+        error(conns@name, conditionMessage(e))
       }
     })
   }
-  .checkLastErrors()
+  if(return_errors == TRUE){
+    cli_text()
+    cli_alert_warning("Error: There are some DataSHIELD errors ")
```

Comment:
What happens when the execution is not in a CLI context, like a batch script execution?

Reply:
Do you mean eg calling datashield.assign.expr in a loop? Or you had something else in mind?

Reply:
No, I mean when there is no command line, like when executing code with Rscript.